### ISSUE TITLE
MPS machine in 3.0 using JAX

### ISSUE THREAD

**Issue by gcarleo** [netket/netket#390]
MPS machines are now written in C++ and at present there is no plan to port them into the pure-python netket 3.0. 

I am pinging here @stavros11, in case he wanted to port them to numpy. 

[Incidentally, this is a much easier task than writing the full class as it was done in C++, since it involves basically writing only the log_val(x) method and the der_log(x), using numpy objects. In other words, there is no look-up table logic and all the other complications that we used to have]

**Comment by gcarleo** [comment]
On a second though, this is much more easily accomplished using JAX. 
Basically one just needs to define a pure python function that multiplies and traces matrices of given bond dimension... the derivatives are then done automatically by JAX. 

If anybody out there is interested in trying to implement this please let us know!

**Comment by tvieijra** [comment]
I could try this, I have some code already using jax for tensor networks.  I think that it is not much work to write something in netket using that.

**Comment by gcarleo** [comment]
Yes! that would be great @tvieijra . I guess the only trickier part is how to randomly initialize the tensors, but I am sure you already though about that...

**Comment by gcarleo** [comment]
Basically this should work as a jax machine, in the sense of what we are finishing up in #396

**Comment by gcarleo** [comment]
@tvieijra just to guide you a bit here, since things have evolved quickly: 

I would suggest that you look at how a simple RBM is implemented in Jax [here](https://github.com/netket/netket/blob/e51df15415f62d93810144321ebb41c1e22c9c68/netket/machine/jax.py#L247) and how it can be used in vmc [here](https://github.com/netket/netket/blob/v3.0/Examples/Jax/jax_example.py) (for the moment SR doesn't work yet, only bare gradient descent). 

Basically, you should be able to write the MPS as a layer-like object, but let us know if you doubts or questions!

**Comment by tvieijra** [comment]
Thank you for the information.  Yes, I think it should be straightforward to write a custom pair of `init_fn` and `forward_fn` for this purpose.  I will get going with it one of these days!
